frontend/src/contexts/authContext.tsx

```diff
@@ -1,14 +1,79 @@
 import type {PropsWithChildren} from 'react';
-import {createContext, useState} from 'react';
+import {createContext, useCallback, useEffect, useState} from 'react';
+import axios from 'axios';
+import * as SecureStore from 'expo-secure-store';
 
-type authContext = { user: { name: string, email: string } | null };
+const setAxiosTokenHeader = (token: string) => axios.defaults.headers.common['authToken'] = `${token}`
+
+const removeAxiosTokenHeader = () => axios.defaults.headers.common['authToken'] = ''
+
+const TOKEN_KEY = 'cba-secure-token';
+type TOKEN_KEY_TYPE = typeof TOKEN_KEY
+
+const setTokenInSecureStore = async (key: TOKEN_KEY_TYPE, value: string) => await SecureStore.setItemAsync(key.trim(), value)
+
+const getTokenFromSecureStore = async (key: TOKEN_KEY_TYPE) => await SecureStore.getItemAsync(key.trim())
+
+const deleteTokenFromSecureStore = async (key: TOKEN_KEY_TYPE) => await SecureStore.deleteItemAsync(key.trim())
+
+
+type userRegistrationData = {
+    name: string,
+    email: string
+}
+type authContext = { 
+    user: { name: string, email: string } | null, 
+    registerUser: (arg: userRegistrationData) => Promise<void>
+    signIn: (email: string) => Promise<void>
+    logOut: () => Promise<void>
+};
 const AuthContext = createContext<authContext | null>(null);
 
 export const AuthContextProvider = ({ children }: PropsWithChildren): JSX.Element => {
     const [user, setUser] = useState(null);
 
+    const registerUser = useCallback(async (data: userRegistrationData) => {
+        try {
+            const { name, email } = data
+        } catch (error) {
+            console.log(error)   
+        }
+    }, [])
+
+    const signIn = useCallback(async (email: string) => {
+        try{
+            let tokenFromBackend = 'tokenFromBackend'
+            setAxiosTokenHeader(tokenFromBackend)
+            await setTokenInSecureStore(TOKEN_KEY, tokenFromBackend);
+        } catch(error) {
+            console.log(error) 
+        }
+    }, [])
+
+    const logOut = useCallback(async () => {
+        try {
+            await deleteTokenFromSecureStore(TOKEN_KEY);
+            removeAxiosTokenHeader()
+            setUser(null)
+        } catch (error) {
+            console.log(error)
+        }
+    }, [])
+
+    useEffect(() => {
```

Comment:
Okay, I think we may talk about this already but if this UseEffect has no dependencies why is the token loading happening inside of it? Can we not just call the function here

Reply:
The function is basically responsible for updating the authentication state whenever the app is opened, which is why I'm doing it in a useEffect with empty dependency array, so that this only happens when the layer is first mounted, which I believe is whenever the app is opened. If there's a different way we could achieve the same outcome, I'm happy to take a look